obspy/io/scardec/core.py

```diff
@@ -0,0 +1,457 @@
+
+    cmt_origin = Origin(
+        resource_id=_get_resource_id(event_name, "origin", tag="cmt"),
+        time=origin_time,
+        longitude=longitude,
+        latitude=latitude,
+        depth=depth,
+        origin_type="centroid",
+        region=_fe.get_region(longitude=longitude,
+                              latitude=latitude)
+    )
+
+    cmt_mag = Magnitude(
+        resource_id=_get_resource_id(event_name, "magnitude", tag="mw"),
+        mag=moment_mag,
+        magnitude_type="mw",
+        origin_id=cmt_origin.resource_id
+    )
+
+    nod1 = NodalPlane(strike=strike1, dip=dip1, rake=rake1)
+    nod2 = NodalPlane(strike=strike2, dip=dip2, rake=rake2)
+    nod = NodalPlanes(nodal_plane_1=nod1, nodal_plane_2=nod2)
+
+    foc_mec = FocalMechanism(
+        resource_id=_get_resource_id(event_name, "focal_mechanism"),
+        nodal_planes=nod
+    )
+
+    dip1 *= np.pi / 180.
+    rake1 *= np.pi / 180.
+    strike1 *= np.pi / 180.
+
+    mxx = - scalar_moment * ((np.sin(dip1) * np.cos(rake1) *
+                              np.sin(2 * strike1)) +
+                             (np.sin(2*dip1) * np.sin(rake1) *
+                              np.sin(2 * strike1)))
+    mxy = scalar_moment * ((np.sin(dip1) * np.cos(rake1) *
+                            np.cos(2 * strike1)) +
+                           (np.sin(2*dip1) * np.sin(rake1) *
+                            np.sin(2 * strike1) * 0.5))
+    myy = scalar_moment * ((np.sin(dip1) * np.cos(rake1) *
+                            np.sin(2 * strike1)) -
+                           (np.sin(2*dip1) * np.sin(rake1) *
+                            np.cos(2 * strike1)))
+    mxz = - scalar_moment * ((np.cos(dip1) * np.cos(rake1) *
+                              np.cos(strike1)) +
+                             (np.cos(2*dip1) * np.sin(rake1) *
+                              np.sin(strike1)))
+    myz = - scalar_moment * ((np.cos(dip1) * np.cos(rake1) *
+                             np.sin(strike1)) -
+                             (np.cos(2*dip1) * np.sin(rake1) *
+                              np.cos(strike1)))
+    mzz = scalar_moment * (np.sin(2*dip1) * np.sin(rake1))
+
+    tensor = Tensor(m_rr=mxx, m_tt=myy, m_pp=mzz, m_rt=mxy, m_rp=mxz, m_tp=myz)
+
+    cm = [Comment(text="Basis system: North,East,Down \
+                        (Jost and Herrmann 1989)")]
+    cm[0].resource_id = _get_resource_id(event_name, 'comment', 'mt')
+    cm.append(Comment(text="MT derived from focal mechanism, therefore \
+                            constrained to pure double couple.",
+                      force_resource_id=False))
+
+    # Write moment rate function
+    extra = {'moment_rate': {'value': stf_mr,
+                             'namespace': r"http://test.org/xmlns/0.1"},
+             'dt': {'value': dt,
+                    'namespace': r"http://test.org/xmlns/0.1"},
+             'offset': {'value': offset,
+                        'namespace': r"http://test.org/xmlns/0.1"}
+             }
```

Comment:
I guess one `SourceTimeFunction` will always only have one single moment rate array? Otherwise `dt` and `offset` could be attributes of `moment_rate`, but I guess having multiple `moment_rate` items would not make sense anyway.